src/scanners/javascript.js

```diff
@@ -23,7 +33,14 @@ export default class JavaScriptScanner {
     this.linterMessages = [];
     this.scannedFiles = [];
     this._rulesProcessed = 0;
-
+    this.disabledRules = typeof options.disabledRules === 'string' ? options.disabledRules.split(',')
+      .reduce((result, i) => {
+        if (!i.trim()) return result;
+        return {
+          ...result,
+          [i.trim()]: true,
+        };
+      }, {}) : {};
```

Comment:
I wonder if a simple `map` (given that `disabledRules` can be an array instead of an object, see comment down below) would work easier here.

```javascript
this.disabledRules = typeof options.disabledRules === 'string' ? options.disabledRules.split(',')
    .map(rule => { rule.trim() })
    .filter(() => { return true; });
```

`map` cleans the strings and `filter` filters out empty elements by default so the result should be similar.

Reply:
it's not the same in my case we have map
```
{ 
  rule0: true,
  rule1: true,
}
```
in your case we have array of rules. it just faster to look in the object if it has property, than lookup in the array if it has item, if you want I can change it to array